package/cloudshell/orch/training/setup_orchestrator.py

```diff
@@ -88,7 +88,7 @@ def register(self, enable_provisioning: bool = True, enable_connectivity: bool =
             self.sandbox.logger.debug("Create user sandboxes logic is added to sandbox orchestration")
             self.sandbox.workflow.on_configuration_ended(self._do_on_configuration_ended, None)
 
-    def _do_on_configuration_ended(self, sandbox, components):
+    def _do_on_configuration_ended(self, sandbox: Sandbox, components):
```

Comment:
Did you test in a live environment?
I think it will fail. The sandbox workflow doesnt support python 3 typing for methods that are added to the workflow...